Test MacOS build

sandboxcoder opened the pull request:
<!--- Please fill out the following template, which will help other contributors review your Pull Request. -->

### Description
Testing the upcoming MacOS release

### Motivation and Context
Obs-studio (MacOS) was recently updated to version 30. Fixing various compilation issues and updating the workflow to copy over dynamic libs to make debugging easier.

### How Has This Been Tested?
Currently testing the binaries against the extensive test suite.

### Types of changes
Fixed various compilation issues
Updated various github workflow scripts where required to copy over .dylibs 

### Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [ ] The code has been tested.
- [ ] All commit messages are properly formatted and commits squashed where appropriate.
- [ ] I have included updates to all appropriate documentation.

sandboxcoder committed:
Reenabling the MacOS build

sandboxcoder committed:
copying over libcurl dylib files (macos build)

sandboxcoder committed:
Fixed macos compiler issues

* Adding osn-multitrack implementation for macos. Currently empty impl but resolves compiler issues
* Windows-only code now separated in CMakeLists

sandboxcoder committed:
macos test libobs version bump

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/signed)](https://cla-assistant.io/streamlabs/obs-studio-node?pullRequest=1517) <br/>All committers have signed the CLA.

sandboxcoder committed:
Copy library files into libobs_src/Framework folder

Now obs64 executable can load the dylib files it requires enabling us to run obs64 from xcode and double click the executable

sandboxcoder committed:
Bump LibOBS version + add module path

* Adding path for macOS Plugins. This resolves the missing "rtmp_service" error because the plugin could not be found

sandboxcoder committed:
Ran clang-format on obs-studio-server/source/nodeobs_api.cpp

sandboxcoder committed:
Renaming .cpp -> .mm file

Renaming this into an obj-C++ file to access hardware info

sandboxcoder committed:
Adding tip on how to attach debugger to OSN tests

sandboxcoder committed:
README.md, adding debugging tests section

sandboxcoder committed:
Adding tip to readme

Tip: how to run specific test

sandboxcoder committed:
Registering macos hotkeys to fix hotkey test

* Adding the new macos hotkeys SCK_AUDIO_CAPTURE & MACOS_AVCAPTURE

sandboxcoder committed:
Removed extraneous comment

sandboxcoder committed:
Adding launch.json to the repo

* So programmers can easily debug the tests from Visual Studio Code

sandboxcoder committed:
README.md indicates how to run specific test

sandboxcoder committed:
Fixed test_osn_input test (macos)

* Refactored the platform independent input types into the shared array.
* each operating system can specify platform dependent types

sandboxcoder committed:
More descriptive var names

sandboxcoder committed:
Disable OPENSSL req by libcurl for mac

Apparently on windows this was already disabled. However, on mac, I am
finding it is necessary to explicitly disable OPENSSL.

sandboxcoder committed:
output debug message for CURL_USE_OPENSSL

sandboxcoder committed:
disable libssh2 req for libcurl

* We currently do not build this binary

sandboxcoder committed:
Install openssl for libcurl

sandboxcoder committed:
disable libssh2 for curl lib

sandboxcoder committed:
Revert "Install openssl for libcurl"

This reverts commit 720712ef3330fef96a4cf6e7c0fafd7f1e847658.
